connectd/connectd.c

```diff
@@ -264,6 +267,9 @@ static struct io_plan *peer_reconnected(struct io_conn *conn,
 	pr->id = *id;
 	pr->cs = *cs;
 	pr->addr = *addr;
+	pr->remote_addr = NULL;
+	if (remote_addr != NULL)
+		pr->remote_addr = remote_addr;
```

Comment:
Isn't a direct assignment pretty much the same here?

Reply:
yup, good catch. don't know how that got in there :D